providers/account.rb

```diff
@@ -169,7 +169,24 @@ def home_ssh_dir_resource(exec_action)
 
 def authorized_keys_resource(exec_action)
   # avoid variable scoping issues in resource block
-  ssh_keys = Array(new_resource.ssh_keys)
+  ssh_keys = []
+  Array(new_resource.ssh_keys).each do |item|
+    # if item is a valid ssh key, copy it.
+    if item =~ /^(ssh-(dss|rsa|ed25519)|ecdsa-sha2-\w+) AAAA/
+      ssh_keys << item
+
+    # if key is not a valid ssh public key, assume it's a username
+    # and try getting the ssh keys from the data bag
+    else
+      user = data_bag_item(new_resource.data_bag, item)
+      if user['keys']
+        ssh_keys += Array(user['keys'])
+      else
+        Chef::Log.info("Couldn't get ssh public keys from data bag '#{new_resource.data_bag}' for user '#{item}'")
```

Comment:
This feels more like a warning and less like an information line. In my mind `INFO` tells you something is happening, while `WARN` is telling you something unexpected may have happened.